### PR TITLE
Pause Pear

### PR DESCRIPTION
**Pull request by vianxnguyen**
## Overview

Implemented pause pear feature, allowing users to temporarily or permanently pause their matches.

## Changes Made

- Completed implementation of PausePearView and implemented UnpausePearView
- Updated NoMatchVC to accommodate various types of pauses and display date of unpause if applicable
- Implemented PausePearFinishView and PausePearFeedbackView 
- Refactored pop up animations.

## Test Coverage

Tested on simulator.

## Next Steps (delete if not applicable)
- Update and integrate Pause Pear Feedback Views

## Screenshots (delete if not applicable)

https://user-images.githubusercontent.com/46759999/165023142-ba948bf2-7429-42a8-9834-c9a7e8eccfe1.mp4

### CODE REVIEW

**Review by mathewscullin — APPROVED** [review]
Love to see it 💯 Some small callouts below but otherwise very nice work!!

File: Pear/Controller/NoMatchViewController.swift [hunk 1]
```diff
@@ -18,10 +18,17 @@ class NoMatchViewController: UIViewController {
     private let availabilityButton = UIButton()
     private let noMatchLabel = UILabel()
     private let noMatchTitleLabel = UILabel()
-    private let surprisedPearImageView = UIImageView()
+    private let pearImageView = UIImageView()
+    private let unpauseButton = DynamicButton()
+    
+    // MARK: - Private Data Vars
+    weak var profileDelegate: ProfileMenuDelegate?
+    private let isPaused: Bool
```
Comment (amyqxhuang) [inline Pear/Controller/NoMatchViewController.swift:26]: alphabetize

File: Pear/Controller/NoMatchViewController.swift [hunk 2]
```diff
@@ -35,44 +42,82 @@ class NoMatchViewController: UIViewController {
         title = "" // To get rid of the "back" text on navigation bar
         view.backgroundColor = .backgroundLightGreen
         navigationController?.navigationBar.isHidden = true
+        setupViews()
+        setupConstraints()
+    }
+    
+    private func getPauseDateString() -> String {
```
Comment (amyqxhuang) [inline Pear/Controller/NoMatchViewController.swift:49]: factor out into Time+Helper

File: Pear/Settings/SettingsViewController.swift [hunk 3]
```diff
@@ -11,29 +11,33 @@ import GoogleSignIn
...
 
     // MARK: - Private Data Vars
     private var settingOptions: [SettingOption] = [
 //        SettingOption(hasSwitch: false, image: "editAvailability", switchOn: false, text: "Edit Time Availabilities"),
 //        SettingOption(hasSwitch: false, image: "location", switchOn: false, text: "Edit Location Availabilities"),
 //        SettingOption(hasSwitch: false, image: "socialMedia", switchOn: false, text: "Connect Social Media"),
         SettingOption(hasSwitch: false, image: "aboutPear", switchOn: false, text: "About Pear"),
-        SettingOption(hasSwitch: false, image: "logout", switchOn: false, text: "Log Out")
+        SettingOption(hasSwitch: false, image: "logout", switchOn: false, text: "Log Out"),
+        SettingOption(hasSwitch: false, image: "pausePear", switchOn: false, text: "Pause Pear")
```
Comment (amyqxhuang) [inline Pear/Settings/SettingsViewController.swift:37]: set the default value for hasSwitch and switchOn to be false in the model

File: Pear/View/PausePearFeedbackView.swift [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    private let pauseLabel = UILabel()
+    private let finishButton = DynamicButton()
+    private let cancelButton = UIButton()
+    private var pauseReasonCollectionView: UICollectionView!
+
+    // MARK: - Private Data Vars
+    private weak var delegate: PausePearDelegate?
+    private var selectedReason: String = ""
+    private let reasonInteritemSpacing: CGFloat = 11
+    private let pauseReasons: [String] = ["Focusing on work", "Won't be on campus", "Bad experience", "Other"]
```
Comment (amyqxhuang) [inline Pear/View/PausePearFeedbackView.swift:23]: alphabetize

File: Pear/View/PausePearFeedbackView.swift [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+import UIKit
+
+class PausePearFeedbackView: UIView {
+
+    // MARK: - Private View Vars
+    private let pauseLabel = UILabel()
+    private let finishButton = DynamicButton()
+    private let cancelButton = UIButton()
+    private var pauseReasonCollectionView: UICollectionView!
```
Comment (amyqxhuang) [inline Pear/View/PausePearFeedbackView.swift:17]: alphabetize

File: Pear/View/PausePearFinishView.swift [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+import UIKit
+
+class PausePearFinishView: UIView {
+
+    // MARK: - Private View Vars
+    private let finishLabel = UILabel()
+    private let feedbackLabel = UILabel()
+    private let feedbackButton = UIButton()
+    private let cancelButton = UIButton()
+    private let pearImageView = UIImageView()
```
Comment (amyqxhuang) [inline Pear/View/PausePearFinishView.swift:18]: alphabetize

File: Pear/View/UnpausePearView.swift [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+//
+
+import UIKit
+
+class UnpausePearView: UIView {
+
+    // MARK: - Private View Vars
+    private let headerLabel = UILabel()
+    private let unpauseButton = DynamicButton()
+    private let cancelButton = UIButton()
```
Comment (amyqxhuang) [inline Pear/View/UnpausePearView.swift:16]: alphabetize

**Review by amyqxhuang — APPROVED** [review]
lgtm! :) just some comments.

also, can "pause pear" be placed above "log out" in the settings table?

File: Pear/Controller/NoMatchViewController.swift [hunk 8]
```diff
@@ -81,16 +126,34 @@ class NoMatchViewController: UIViewController {
...
+            make.top.equalTo(noMatchTitleLabel.snp.bottom).inset(imageBottomPadding)
+            make.bottom.equalTo(unpauseButton.snp.top).offset(imageBottomPadding)
+            make.leading.trailing.equalToSuperview().inset(pearImageViewPadding)
+        }
+        
+        unpauseButton.snp.makeConstraints { make in
             make.centerX.equalToSuperview()
-            make.bottom.equalTo(noMatchLabel.snp.top).offset(-imageBottomPadding)
-            make.width.equalTo(imageWidth)
+            make.bottom.equalTo(noMatchLabel.snp.top).offset(-subtitleLabelPadding)
```
[the remark this answers is not in the document]
Comment (vianxnguyen) [reply Pear/Controller/NoMatchViewController.swift:141]: not sure why, but when i tried that, the constraints ended up looking inconsistent

**Pull request merged**